extensions/amp-sticky-ad/1.0/amp-sticky-ad.js

```diff
@@ -69,28 +70,24 @@ class AmpStickyAd extends AMP.BaseElement {
 
     this.ad_ = children[0];
     this.setAsOwner(this.ad_);
-
-    // TODO(@zhouyx, #9126): cleanup once research
-    // on custom-element stubbing is complete.
-    let customApiResolver;
-    const customApiPromise = new Promise(resolve => {
-      customApiResolver = resolve;
-    });
-    if (this.ad_.whenBuilt) {
-      customApiResolver();
-    } else {
-      // Give 1s for amp-ad to stub. Report 1% error.
-      if (Math.random() < 0.01) {
-        dev().error(TAG, 'race condition on customElement stubbing');
-      }
-      timerFor(this.win).delay(customApiResolver, 1000);
-    }
-    customApiPromise.then(() => {
-      this.ad_.whenBuilt().then(() => {
-        this.mutateElement(() => {
-          toggle(this.element, true);
+    whenUpgradedToCustomElement(dev().assertElement(this.ad_)).then(ad => {
+      if (!ad.whenBuilt) {
+        // TODO(@zhouyx, #9126): Cleanup once make sure the fix works
+        dev().error(TAG, 'element whenBuilt still do not exist!');
+        timerFor(this.win).delay(() => {
```

Comment:
haha, I think @dvoytenko is right. it's not needed. I already tested it and am pretty sure it fix the race condition bug in #9126. @jridgewell I deleted the duplication code 😄